### DISCUSSION TITLE
sql contain "with as" , how get tablename

### DISCUSSION THREAD

**Discussion by zbjbaggio** [apache/doris#12005] [A - General / Q&A]
for example:
with test_customer as(
                select c_customer_sk,c_customer_id,c_current_cdemo_sk
                from tpc_db.customer as q
                where c_customer_sk
                in (
                select c_customer_sk
                from tpc_db.customer
                where c_salutation='test'
                group by 1 having COUNT(DISTINCT c_salutation) >4)
                )
                select * from test_customer limit 10

**Comment by zbjbaggio** [comment]
SelectStmt selectStmt = (SelectStmt) queryStmt;;
            List<TableRef> tableRefs = selectStmt.getTableRefs();
            This methond get test_customer  is not  tpc_db.customer

**Reply by morningman** [reply]
If you want to get real table name for a given sql in code, you can refer to this PR:
https://github.com/apache/doris/pull/11979
